### DISCUSSION TITLE
IPOPT optimal solution found while constraints are violated

### DISCUSSION THREAD

**Discussion by elh-mod1360** [coin-or/Ipopt#440] [Q&A] [answered]
I am solving a nonlinear problem with several equality and in equality constraints. The problem is that IPOT says that the optimal solution is found while the constraints on derivatives of some control variables are violated. I have also decreased the option: constr_viol_tol,  but still get the same result.
I really appreciate your help in advance.
Best Regards,
Elham.

**Comment by svigerske** [comment, marked as answer]
Try setting honor_original_bounds=no. This should prevent that Ipopt projects the final point back onto the given variable bounds. The returned point will then have a constraint violation that should satisfy the constr_viol_tol, but that may be outside variable bounds by a bit.
If that is a problem, try setting bound_relax_factor=0.
With Ipopt 3.14, honor_original_bounds=no is the default and relaxation of bounds is limited by constr_viol_tol in addition.
